### DISCUSSION TITLE
Get markdown content from Database

### DISCUSSION THREAD

**Discussion by adityar15** [nuxt/content#1849] [Q&A]
Hi I am using Nuxt 3 with @nuxt/content v2

I am having an admin dashboard from where users can write the markdown and the markdown content is stored in the database.

Now I want to render this markdown content on my Nuxt 3 website. What's the best way to do it and do I need to use <ContentRenderer /> component?

How can I make sure that the code highlighting is not disturbed?

Thanks in advance :)

**Comment by sudomk** [comment]
Hi, 
I am also interested in what other people say about that. 

I already tried a similar approach by rendering a markdown file to an AST with `nuxt/content` and storing that AST in a database. Then on another page (or even another application) I fetched the AST and let it render by the `<ContentRenderer />` component. This works so far but is kind of ugly as the AST may contain Vue components that could be present in a context where the AST was generated, but not in the context where it gets rendered.

Another  approach could be to configure a foreign source for content files (see https://content.nuxtjs.org/api/configuration#sources). Unfortunately, the documentation is still very sparse with information about how to configure different sources properly. But apparently it's just the configuration of `unstorage`, which I'm unfamiliar with.

**Comment by tveimo** [comment]
Writing a driver for nuxt content is not too hard. Here's some sample drivers;

https://github.com/unjs/unstorage/tree/main/src/drivers

Some documentation on unstorage custom drivers, the storage framework used in nuxt content; https://unstorage.unjs.io/custom-driver

Someone trying out writing a driver here; https://github.com/bloodf/nuxt_content_lazyload_driver/blob/main/drivers/devTo/driver.mjs
